amazon/ion/writer_binary_raw.py

```diff
@@ -185,7 +185,11 @@ def _serialize_decimal(ion_event):
     value = ion_event.value
     validate_scalar_value(value, Decimal)
     sign, digits, exponent = value.as_tuple()
-    coefficient = int(value.scaleb(-exponent).to_integral_value())
+    with localcontext() as context:
+        # Adjusting precision for taking into account arbitrarily large/small
+        # numbers
+        context.prec = len(digits) + abs(exponent)
```

Comment:
Similarly, this can simply be `context.prec = len(digits)`

Reply:
Changes made.